From 7ba6a9e261654fab3ecc8e5b939e02028859cec8 Mon Sep 17 00:00:00 2001
From: Spencer Powell <krornus@gmail.com>
Date: Fri, 16 Sep 2022 13:38:06 -0600
Subject: [PATCH 2/8] Add .gitignore

---
 .gitignore | 6 ++++++
 1 file changed, 6 insertions(+)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..e2ec094
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,6 @@
+*.orig
+*.rej
+*.o
+*.d
+config.h
+dwm
-- 
2.37.2

